### DISCUSSION TITLE
Linking parameters of two instruments

### DISCUSSION THREAD

**Discussion by aaronsharpe** [microsoft/Qcodes#3512] [Q&A]
I was wondering of a more proper way to link parameters of two different instruments according to some relation.

This is an example of how we would hack something like this in the past. k1 and k2 are something like Keithley 2400s. 
![example](https://user-images.githubusercontent.com/24236207/138976786-cbfcfb08-c6a4-472a-b262-488e151c13ae.png)

**Comment by FarBo** [comment]
Hi aaron,

Thank you for your question. This is indeed something that is missing in qcodes and we have a workflow to implement such a feature, but we cannot promise any timeline for that.

For now, what you have propsed as "#orthis" is the way to go, despite its limitation that if you want to go back to the dataset that is created by this hack after a while, you don't have any information of the logic "twokeefs_fn" function stored in the dataset.
you should, however, make sure to add this parameter to the station object (I assume you have a station created before) by `station.add_component(twokeefs)`.

There is perhaps another way, and that's defining a virtual qcodes instrument class, and setup the logic there.
